backend/app/controllers/uoas.js

```diff
@@ -157,40 +159,115 @@ module.exports = {
+                        );
+
+                        if (!_.first(task)) {
+                            // Soft delete the UOA from the Product UAO Table
+                            yield thunkQuery(
+                                'UPDATE "ProductUOA"' +
+                                'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                                '/ 1000.0)) WHERE "UOAid" = ' + req.params.id +
+                                'AND "productId" = ' + req.body.productId
+                            );
+
+                            // Soft delete from the UOA table
+                            yield thunkQuery(
+                                'UPDATE "UnitOfAnalysis"' +
+                                'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                                '/ 1000.0)) WHERE "id" = ' + req.params.id
+                            );
+
+                        } else {
+                            if (task[0].isComplete === true) {
+                                throw new HttpError(403, 'Cannot delete UOA of already completed task');
+                            } else {
+                                // TODO: Add check to make sure there aren't answered questions for survey
+                                console.log(`FOUND A TASK`)
+                                // Soft delete the UOA from the Product UAO Table
+                                yield thunkQuery(
+                                    'UPDATE "ProductUOA"' +
+                                    'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                                    '/ 1000.0)) WHERE "UOAid" = ' + req.params.id +
+                                    'AND "productId" = ' + req.body.productId
+                                );
+
+                                // Soft delete the task with that UAO ID
+                                yield thunkQuery(
+                                    'UPDATE "Tasks"' +
+                                    'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                                    '/ 1000.0)) WHERE "productId" = ' + req.body.productId +
+                                    'AND "uoaId" = ' + req.params.id
+                                );
+
+                                yield thunkQuery(
+                                    'UPDATE "UnitOfAnalysis"' +
+                                    'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                                    '/ 1000.0)) WHERE "id" = ' + req.params.id
+                                );
+                            }
+                        }
+                    }
                 } else {
-                    throw new HttpError(403, 'Cannot delete UOA of already completed task');
+                    throw new HttpError(403, 'No Project Found');
                 }
             } else {
+                // TODO: Replicate checks for conditions if there is no productID passed in.
```

Comment:
Is the work referred to by this TODO going to be done under a separate ticket?

Reply:
Yes